pkg/controller/installation/products/codeready/reconciler.go

```diff
@@ -0,0 +1,385 @@
+	}
+
+	// retrieve the sso config so we can find the keycloakrealm custom resource to update
+	kcCfg, err := r.ConfigManager.ReadRHSSO()
+	if err != nil {
+		return v1alpha1.PhaseFailed, err
+	}
+	if kcCfg.GetRealm() == "" {
+		return v1alpha1.PhaseFailed, errors.New("keycloak config Realm is not set")
+	}
+	kcRealm := &keycloakv1.KeycloakRealm{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      kcCfg.GetRealm(),
+			Namespace: kcCfg.GetNamespace(),
+		},
+	}
+	serverClient.Get(ctx, pkgclient.ObjectKey{Name: kcCfg.GetRealm(), Namespace: kcCfg.GetNamespace()}, kcRealm)
+
+	// Create a che client that can be used in keycloak for che to login with
+	kcCheClient := &keycloakv1.KeycloakClient{}
+	if existingKcCheClient := findCheKeycloakClient(kcRealm.Spec.Clients); existingKcCheClient != nil {
+		kcCheClient = existingKcCheClient
+	}
+	kcCheClient.KeycloakApiClient = &keycloakv1.KeycloakApiClient{
+		ID:                        defaultClientName,
+		Name:                      defaultClientName,
+		Enabled:                   true,
+		StandardFlowEnabled:       true,
+		DirectAccessGrantsEnabled: true,
+		RootURL:                   cheURL,
+		RedirectUris:              []string{cheURL, fmt.Sprintf("%s/*", cheURL)},
+		AdminURL:                  cheURL,
+		WebOrigins:                []string{cheURL, fmt.Sprintf("%s/*", cheURL)},
+		PublicClient:              true,
+		ClientAuthenticatorType:   "client-secret",
+	}
+
+	// Append the che client to the list of clients in the keycloak realm and save
+	kcRealm.Spec.Clients = findOrAppendCheKeycloakClient(kcRealm.Spec.Clients, kcCheClient)
+	err = serverClient.Update(ctx, kcRealm)
+	if err != nil {
+		return v1alpha1.PhaseFailed, pkgerr.Wrap(err, "could not update keycloakrealm custom resource with codeready client")
+	}
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func findOrAppendCheKeycloakClient(clients []*keycloakv1.KeycloakClient, toAppend *keycloakv1.KeycloakClient) []*keycloakv1.KeycloakClient {
+	if existingKcCheClient := findCheKeycloakClient(clients); existingKcCheClient != nil {
+		return clients
+	}
+	return append(clients, toAppend)
+}
+
+func findCheKeycloakClient(clients []*keycloakv1.KeycloakClient) *keycloakv1.KeycloakClient {
+	for _, client := range clients {
+		if client.ID == defaultClientName {
+			return client
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) createCheCluster(ctx context.Context, serverClient pkgclient.Client, kcCfg *config.RHSSO, kr *keycloakv1.KeycloakRealm) error {
+	cheCluster := &chev1.CheCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      defaultCheClusterName,
+			Namespace: r.Config.GetNamespace(),
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf(
+				"%s/%s",
+				chev1.SchemeGroupVersion.Group,
+				chev1.SchemeGroupVersion.Version,
+			),
+			Kind: "CheCluster",
+		},
+		Spec: chev1.CheClusterSpec{
+			Server: chev1.CheClusterSpecServer{
+				CheFlavor:      "codeready",
+				TlsSupport:     true,
+				SelfSignedCert: false,
```

Comment:
should this be an option as if we are using RHPDS it may be self signed

Reply:
I'll add an attribute to the `Installation` spec for this